pkg/keystone/fernet.go

```diff
@@ -16,15 +16,22 @@ limitations under the License.
 package keystone
 
 import (
+	"crypto/rand"
 	"encoding/base64"
-	"math/rand"
+
+	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
-// GenerateFernetKey -
+// GenerateFernetKey - returns a base64-encoded, 32-byte key using cryptographically secure random generation
 func GenerateFernetKey() string {
+	logger := log.Log.WithName("GenerateFernetKey")
```

Comment:
maybe just pass the logger `r.GetLogger(ctx)` to GenerateFernetKey() instead of creating each time a new one?